core/src/main/java/org/ehcache/core/config/ResourcePoolsImpl.java

```diff
@@ -100,14 +102,30 @@ public ResourcePools validateAndMerge(ResourcePools toBeUpdated) {
    * @param pools the resource pools to validate
    */
   public static void validateResourcePools(Collection<? extends ResourcePool> pools) {
-    EnumMap<ResourceType.Core, SizedResourcePool> coreResources = new EnumMap<ResourceType.Core, SizedResourcePool>(ResourceType.Core.class);
-    for (ResourcePool pool : pools) {
-      if (pool.getType() instanceof ResourceType.Core) {
-        coreResources.put((ResourceType.Core)pool.getType(), (SizedResourcePool)pool);
+    List<SizedResourcePool> ordered = new ArrayList<SizedResourcePool>(pools.size());
+    for(ResourcePool pool : pools) {
+      if (pool instanceof SizedResourcePool) {
+        ordered.add((SizedResourcePool)pool);
       }
     }
+    Collections.sort(ordered, new Comparator<SizedResourcePool>() {
+      @Override
+      public int compare(final SizedResourcePool o1, final SizedResourcePool o2) {
+        int retVal = o2.getType().getTierHeight() - o1.getType().getTierHeight();
+        if(retVal == 0) {
```

Comment:
This looks like checking for duplicate tier height. I find it weird to do that inline with the sorting. I would rather do it when we walk the sorted list to assess tier inversion.